roxie/ccd/ccdqueue.cpp

```diff
@@ -1489,9 +1549,10 @@ class RoxieReceiverBase : implements IRoxieOutputQueueManager, public CInterface
             assertex(subChannels[channel] == 0);
             assertex(subChannel != 0);
             subChannels[channel] = subChannel;
-            liveSubChannels[channel] = 0;  // Until proven otherwise
             channels[channelCount++] = channel;
         }
+        for (int i = 0; i < channelCount; i++)
+            subChannelInfo[i].init(i);
```

Comment:
subChannelInfo[channels[i]].init(channels[i]);

or merge into the previous loop